integration/integration_test.go

```diff
@@ -1163,89 +1148,71 @@ func testEscapeSequenceNoTrigger(t *testing.T, terminal *Terminal, sess <-chan e
+		Host:    Host,
+	})
+	require.NoError(t, err)
+
 	personA := NewTerminal(250)
-	personB := NewTerminal(250)
+	clientA.Stdout = personA
+	clientA.Stdin = personA
 
-	// PersonA: SSH into the server, wait one second, then type some commands on stdin:
 	sessionA := make(chan error)
-	openSession := func() {
-		cl, err := teleport.NewClient(helpers.ClientConfig{
-			Login:   username,
-			Cluster: helpers.Site,
-			Host:    Host,
-		})
-		if err != nil {
-			sessionA <- trace.Wrap(err)
-			return
-		}
-		cl.Stdout = personA
-		cl.Stdin = personA
-		// Person A types something into the terminal (including "exit")
-		personA.Type("\aecho hi\n\r\aexit\n\r\a")
-		sessionA <- cl.SSH(context.TODO(), []string{}, false)
-	}
+	go func() {
+		sessionA <- clientA.SSH(ctx, []string{}, false)
+	}()
 
-	// PersonB: wait for a session to become available, then join:
-	sessionB := make(chan error)
-	joinSession := func() {
-		sessionTimeoutCtx, sessionTimeoutCancel := context.WithTimeout(context.Background(), 10*time.Second)
-		defer sessionTimeoutCancel()
-		sessions, err := waitForSessionToBeEstablished(sessionTimeoutCtx, apidefaults.Namespace, site)
-		if err != nil {
-			sessionB <- trace.Wrap(err)
-			return
-		}
+	// wait for the session to become available
+	sess := waitForSessionToBeEstablished(ctx, t, site)
+	sid := sess.GetSessionID()
 
-		sessionID := string(sessions[0].ID)
-		cl, err := teleport.NewClient(helpers.ClientConfig{
-			Login:   username,
-			Cluster: helpers.Site,
-			Host:    Host,
-		})
-		if err != nil {
-			sessionB <- trace.Wrap(err)
-			return
-		}
+	// PersonB: join the session
+	clientB, err := teleport.NewClient(helpers.ClientConfig{
+		Login:   username,
+		Cluster: helpers.Site,
+		Host:    Host,
+	})
+	require.NoError(t, err)
 
-		timeoutCtx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-		defer cancel()
+	personB := NewTerminal(250)
+	clientB.Stdout = personB
+	clientB.Stdin = personB
 
-		ticker := time.NewTicker(100 * time.Millisecond)
-		defer ticker.Stop()
-		for {
-			select {
-			case <-timeoutCtx.Done():
-				sessionB <- timeoutCtx.Err()
-				return
+	sessionB := make(chan error)
+	go func() {
+		sessionB <- clientB.Join(ctx, types.SessionPeerMode, apidefaults.Namespace, session.ID(sid), personB)
+	}()
 
-			case <-ticker.C:
-				err := cl.Join(context.TODO(), types.SessionPeerMode, apidefaults.Namespace, session.ID(sessionID), personB)
-				if err == nil {
-					sessionB <- nil
-					return
-				}
+	// Person A types something into the terminal (including "exit")
+	personA.Type("\aecho hi\n\r\aexit\n\r\a")
+
+	// wait for the sessions to end without error
+	sessionEndedWithoutError := func(t *testing.T, errC chan error) func() bool {
+		return func() bool {
+			select {
+			case err := <-errC:
+				require.NoError(t, err)
+				return true
+			default:
+				return false
 			}
 		}
 	}
+	require.Eventually(t, sessionEndedWithoutError(t, sessionA), time.Second*10, time.Second)
```

Comment:
Can you keep polling every 250ms or every 500ms? 1-second timeout one after another will extend the overall test execution time.